ArduPlane/soaring.cpp

```diff
@@ -0,0 +1,96 @@
+#include "Plane.h"
+
+/*
+*  ArduSoar support function
+*
+*  Peter Braswell, Samuel Tabor, Andrey Kolobov, and Iain Guilliard
+*/
+void Plane::update_soaring() {
+    
+    if (!g2.soaring_controller.is_active()) {
+        return;
+    }
+    
+    g2.soaring_controller.update_vario();
+
+    // Check for throttle suppression change.
+    switch (control_mode){
+    case AUTO:
+        g2.soaring_controller.suppress_throttle();
+        break;
+    case FLY_BY_WIRE_B:
+    case CRUISE:
```

Comment:
I'll defer to @tridge as to what his idea for CRUISE mode implementation is, but the behavior used to be that when we reach SOAR_ALT_MIN, we enable throttle and fly back, correct? If this is the case, then we should preserve it the way it was. 

In general, we should really stop adding non-critical changes to this PR. We are running unjustified risks of introducing new bugs for no gain -- all these tweaks can just as well be added after the PR goes through.